### PR TITLE
Add Inngest guide

### PR DESCRIPTION
**Pull request by djfarrelly**
Adds a guide for using Clerk's new webhook integration with Inngest.

This guide aims to show how Clerk webhook events can be used for much more than just syncing data. Clerk can be the hub of all user management events triggering flows from marketing to trials and billing. This guide shows how developers can do that, leveraging Inngest to help in the process. 

Aimed to emulate the tone of existing Clerk documentation.

Submitting docs directly here after convo w/ @domitriusclark and Nick Parsons.

--- 

_Additional context_: This guide may be more appropriate along side the "Sync Data" guide in the users section as it shows how a developer can use Clerk webhooks for syncing, but a lot more. I decided to submit in the integrations section as that's also where the info for Webhooks is. Very happy to make adjustments or link to this guide from the "Sync Data" guide as well.

### CODE REVIEW

File: docs/integrations/webhooks/inngest.mdx [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+To quickly get events to test within Dev Server, you can select any individual event from the Inngest Events tab then select the **Send to Dev Server**.
+
+<Images
+  width={2142}
+  height={1246}
+  src="/docs/images/integrations/inngest/inngest-send-to-dev-server.webp"
+  alt="The Inngest Dashboard showing an individual event payload. Red arrows point to the Events tab and the Send to Dev Server button."
+/>
+
+You'll now see the event in the Inngest Dev Server's stream tab alongside any functions that it triggered.
```
Comment (sylwiavargas) [inline docs/integrations/webhooks/inngest.mdx:197]:
```suggestion
You'll now see the event in the Inngest Dev Server's **Stream** tab alongside any functions that it triggered.
```

File: docs/integrations/webhooks/inngest.mdx [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+
+During local development with Inngest, you can use the Inngest Dev Server to run and test your functions on your own machine. To start the server, in your project directory run the following command:
+
+```sh filename="terminal"
+npx inngest-cli@latest dev
+```
+
+In your browser open [http://localhost:8288](http://localhost:8288) to see the development UI.
+
+To quickly get events to test within Dev Server, you can select any individual event from the Inngest Events tab then select the **Send to Dev Server**.
```
Comment (sylwiavargas) [inline docs/integrations/webhooks/inngest.mdx:188]:
```suggestion
To quickly get events to test within Dev Server, you can select any individual event from the **Events** tab then select the **Send to Dev Server**.
```

File: docs/integrations/webhooks/inngest.mdx [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+ }
+)
+```
+
+Now, you've extended the usefulness of Clerk webhook events even further to build an onboarding drip email campaign in just a few lines of code.
+
+
+## Testing webhook events using the Inngest Dev Server
+
+During local development with Inngest, you can use the Inngest Dev Server to run and test your functions on your own machine. To start the server, in your project directory run the following command:
```
Comment (sylwiavargas) [inline docs/integrations/webhooks/inngest.mdx:180]:
```suggestion
During local development with Inngest, you can use the [Inngest Dev Server](https://www.inngest.com/docs/local-development) to run and test your functions on your own machine. To start the server, in your project directory run the following command:
```

File: docs/integrations/webhooks/inngest.mdx [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+
+
+Now, you have a function that utilizes the same Clerk webhook event for another purpose. Clerk webhook events can be used for all sorts of application lifecycle use cases. For example, adding users to a marketing email list, starting a Stripe trial, or provisioning new account resources.
+
+
+### Sending a delayed follow-up email
+
+Every Inngest function handler has an additional `step `object which provides tools to create more complex functions. Using `step.run` allows you to encapsulate specific code that will be automatically retried ensuring that issues with one part of your function don't force the entire function to re-run. Additionally, [other tools like `step.sleep`](https://www.inngest.com/docs/reference/functions/step-sleep), are available to extend functionality.
+
+The code below uses `step.sleep `to send a welcome email, then wait for three days, and follow up with a message offering a free trial:
```
Comment (sylwiavargas) [inline docs/integrations/webhooks/inngest.mdx:147]:
```suggestion
The code below uses `step.sleep `to send a welcome email, then wait for three days, and follows up with a message offering a free trial:
```

File: docs/integrations/webhooks/inngest.mdx [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+)
+```
+
+
+Now, you have a function that utilizes the same Clerk webhook event for another purpose. Clerk webhook events can be used for all sorts of application lifecycle use cases. For example, adding users to a marketing email list, starting a Stripe trial, or provisioning new account resources.
+
+
+### Sending a delayed follow-up email
+
+Every Inngest function handler has an additional `step `object which provides tools to create more complex functions. Using `step.run` allows you to encapsulate specific code that will be automatically retried ensuring that issues with one part of your function don't force the entire function to re-run. Additionally, [other tools like `step.sleep`](https://www.inngest.com/docs/reference/functions/step-sleep), are available to extend functionality.
```
Comment (sylwiavargas) [inline docs/integrations/webhooks/inngest.mdx:145]:
```suggestion
Every Inngest function handler has an additional `step` object which provides tools to create more complex functions. Using `step.run` allows you to encapsulate specific code that will be automatically retried ensuring that issues with one part of your function don't force the entire function to re-run. Additionally, [other tools like `step.sleep`](https://www.inngest.com/docs/reference/functions/step-sleep), are available to extend functionality.
```

File: docs/integrations/webhooks/inngest.mdx [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+As you can see, you can choose which events you want to handle with each function. You might write a separate function for `clerk/user.updated` and `clerk/user.deleted` handling the entire lifecycle end to end.
+
+Note that multiple functions can also listen to the same event. This pattern is called “[fan-out](https://www.inngest.com/docs/guides/fan-out-jobs).”
+
+
+## Creating a function to send a welcome email
+
+Often, applications need to perform additional tasks when a new user is created, like send a welcome email with tips and useful information.
+
+While it is possible to add this logic at the end of your sync function as seen in the previous section, it's better to decouple logic into different functions so issues with one task do not affect the other ones. For example, if your email fails to send, it should not affect adding the user to an email list.
```
Comment (sylwiavargas) [inline docs/integrations/webhooks/inngest.mdx:117]:
```suggestion
While it is possible to add this logic at the end of your sync function as seen in the [previous section](/docs/integrations/webhooks/inngest#creating-a-function-to-sync-a-new-user-to-a-database), it's better to decouple logic into different functions so issues with one task do not affect the other ones. For example, if your email fails to send, it should not affect adding the user to an email list.
```

File: docs/integrations/webhooks/inngest.mdx [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+Clerk dashboard. Next, select the **Add Endpoint** button.
+
+<Images
+  width={2578}
+  height={1652}
+  src="/docs/images/integrations/inngest/webhook-page.webp"
+  alt="The Webhooks page in the Clerk Dashboard. A red arrow points to the button for Add Endpoint."
+/>
+
+On the next page, select the **Transformation template** tab and the **Inngest** template, then click on the **Connect to Inngest **button**.**
```
Comment (sylwiavargas) [inline docs/integrations/webhooks/inngest.mdx:32]:
```suggestion
On the next page, select the **Transformation template** tab and the **Inngest** template, then click on the **Connect to Inngest** button.
```

File: docs/integrations/webhooks/inngest.mdx [hunk 8]
```diff
@@ -0,0 +1,217 @@
...
+
+In this guide, you'll learn how to set up Inngest to receive Clerk webhook events and how to define Inngest functions in your application using Clerk events including synchronizing data and sending welcome emails.
+
+To follow this guide, you need an Inngest account and have Inngest [set up](https://www.inngest.com/docs/quick-start) in your codebase.
+
+## Setting up the Inngest webhook
+
+To create an Inngest webhook endpoint and add it to your Clerk account, open the [Webhooks](https://dashboard.clerk.com/last-active?path=webhooks) page at the
+
+Clerk dashboard. Next, select the **Add Endpoint** button.
```
Comment (sylwiavargas) [inline docs/integrations/webhooks/inngest.mdx:23]:
```suggestion
To create an Inngest webhook endpoint and add it to your Clerk account, open the [Webhooks](https://dashboard.clerk.com/last-active?path=webhooks) page at the Clerk dashboard. Next, select the **Add Endpoint** button.
```

File: docs/integrations/webhooks/inngest.mdx [hunk 9]
```diff
@@ -0,0 +1,217 @@
...
+---
+
+# Handling webhooks with Inngest
+
+Webhooks allow you to [synchronize data](/docs/users/sync-data) from Clerk to your application backend. You can either handle them directly in your backend with an endpoint or use a tool like [Inngest](https://inngest.com/) which receives the webhook events for you and reliably executes functions in your codebase. When handling webhooks, Inngest receives the webhook events for you and uses a built-in queue to reliably execute longer running functions with additional functionality including:
+
+* [Limiting concurrency](https://www.inngest.com/docs/guides/concurrency) to handle spikes in events without overwhelming your API or database.
+* Trigger multiple functions from a single event ([fan-out jobs](https://www.inngest.com/docs/guides/fan-out-jobs)).
+* [Delay code](https://www.inngest.com/docs/functions/multi-step) to run after a period of time.
+* [Debounce events](https://www.inngest.com/docs/reference/functions/debounce) to reduce duplicate processing.
```
Comment (sylwiavargas) [inline docs/integrations/webhooks/inngest.mdx:13]:
```suggestion
* Triggering multiple functions from a single event ([fan-out jobs](https://www.inngest.com/docs/guides/fan-out-jobs)).
* [Delaying code](https://www.inngest.com/docs/functions/multi-step) to run after a period of time.
* [Debouncing events](https://www.inngest.com/docs/reference/functions/debounce) to reduce duplicate processing.
```

File: docs/integrations/webhooks/inngest.mdx [hunk 10]
```diff
@@ -0,0 +1,215 @@
...
+/>
+
+A popup window will appear to complete the setup. Select **Approve** to create the webhook.
+
+<Images
+  width={1880}
+  height={1448}
+  src="/docs/images/integrations/inngest/webhook-endpoint-connected.webp"
+  alt="The Webhooks page in the Clerk Dashboard showing a connected Inngest account. A red arrow points to the Connected button."
+/>
```
Comment (royanger) [inline docs/integrations/webhooks/inngest.mdx:46]: This image appears to be in the wrong order. Looks like it should be flipped with the one below it. 
```suggestion
<Images
  width={1194}
  height={1520}
  src="/docs/images/integrations/inngest/inngest-permissions-dialog.webp"
  alt="The Inngest permissions popup window showing the Approve button."
/>
```

File: docs/integrations/webhooks/inngest.mdx [hunk 11]
```diff
@@ -0,0 +1,215 @@
...
+
+After the popup window disappears, the Webhooks page will now display **Connected** with the webhook URL underneath. There is one more step to complete setup.
+
+<Images
+  width={1194}
+  height={1520}
+  src="/docs/images/integrations/inngest/inngest-permissions-dialog.webp"
+  alt="The Inngest permissions popup window showing the Approve button."
+/>
```
Comment (royanger) [inline docs/integrations/webhooks/inngest.mdx:55]:
```suggestion
<Images
  width={1880}
  height={1448}
  src="/docs/images/integrations/inngest/webhook-endpoint-connected.webp"
  alt="The Webhooks page in the Clerk Dashboard showing a connected Inngest account. A red arrow points to the Connected button."
/>
```

File: docs/integrations/webhooks/inngest.mdx [hunk 12]
```diff
@@ -0,0 +1,215 @@
...
+  alt="The Webhooks page in the Clerk Dashboard showing the end of the page to create a new endpoint. A red arrow points to the Create button."
+/>
+
+
+You'll be redirected to the new endpoint. In your Inngest dashboard, you will see a new webhook created in your account's [production environment](https://app.inngest.com/env/production/manage/webhooks).
+
+
+## Viewing webhook events within Inngest
+
+After setup, as webhook events are sent from Clerk to Inngest, new `clerk `events will appear  in your Inngest dashboard. Event names will be the prefixed `clerk/` followed by the event name. See [webhook events](/docs/integrations/webhooks#supported-webhook-events) for a full list.
```
Comment (royanger) [inline docs/integrations/webhooks/inngest.mdx:72]: Maybe link 'Inngest dashboard' to the dashboard so it is easy to view?

File: docs/integrations/webhooks/inngest.mdx [hunk 13]
```diff
@@ -0,0 +1,215 @@
...
+ async ({ event }) => {
+   const user = event.data; // The event payload's data will be the Clerk User json object
+   const { id, first_name, last_name } = user;
+   const email = user.email_addresses.find(e =>
+     e.id === user.primary_email_address_id
+   ).email
+   await database.users.insert({ id, email, first_name, last_name })
+ }
+)
+```
```
Comment (royanger) [inline docs/integrations/webhooks/inngest.mdx:101]: I think we can make this a little clearer that this in the customer's codebase. You can use something like `filename="src/inngest/sync-user.ts"` on line 88 after the language as well. 

Also, having not used Inngest and just reading this I am wondering if the 'id' value is just a label as it appears to be or if it should be an id string of some sort.

File: docs/integrations/webhooks/inngest.mdx [hunk 14]
```diff
@@ -0,0 +1,215 @@
+---
+title: Handling webhooks with Inngest
+description: Learn how to integrate Clerk with Inngest.
+---
+
+# Handling webhooks with Inngest
+
+Webhooks allow you to [synchronize data](/docs/users/sync-data) from Clerk to your application backend. You can either handle them directly in your backend with an endpoint or use a tool like [Inngest](https://inngest.com/) which receives the webhook events for you and reliably executes functions in your codebase. When handling webhooks, Inngest receives the webhook events for you and uses a built-in queue to reliably execute longer running functions with additional functionality including:
```
Comment (royanger) [inline docs/integrations/webhooks/inngest.mdx:8]:
```suggestion
Webhooks allow you to [synchronize data](/docs/users/sync-data) from Clerk to your application backend. You can either handle them directly in your backend with an endpoint or use a tool like [Inngest](https://inngest.com/) which receives the webhook events for you and reliably executes functions in your codebase. When handling webhooks, Inngest receives the [webhook events](/docs/integrations/webhooks#supported-webhook-events) for you and uses a built-in queue to reliably execute longer running functions with additional functionality including:
```

File: docs/integrations/webhooks/inngest.mdx [hunk 15]
```diff
@@ -0,0 +1,215 @@
...
+
+
+Now, you have a function that utilizes the same Clerk webhook event for another purpose. Clerk webhook events can be used for all sorts of application lifecycle use cases. For example, adding users to a marketing email list, starting a Stripe trial, or provisioning new account resources.
+
+
+### Sending a delayed follow-up email
+
+Every Inngest function handler has an additional `step` object which provides tools to create more complex functions. Using `step.run` allows you to encapsulate specific code that will be automatically retried ensuring that issues with one part of your function don't force the entire function to re-run. Additionally, [other tools like `step.sleep`](https://www.inngest.com/docs/reference/functions/step-sleep), are available to extend functionality.
+
+The code below sends a welcome email, then uses `step.sleep` to wait for three days before sending another email offering a free trial:
```
Comment (royanger) [inline docs/integrations/webhooks/inngest.mdx:145]: In the previous section the guide indicates to decouple tasks so they don't affect each other, but this example has two tasks coupled together. Is there a reason for this? It came off weird reading the guide going from the previous section to this one section.

File: docs/integrations/webhooks/inngest.mdx [hunk 16]
```diff
@@ -0,0 +1,215 @@
...
+const sendWelcomeEmail = inngest.createFunction(
+ { id: 'send-welcome-email' },
+ { event: 'clerk/user.created' },
+ async ({ event }) => {
+   const { user } = event.data
+   const { first_name } = user
+   const email = user.email_addresses.find(e =>
+     e.id === user.primary_email_address_id
+   ).email
+   await emails.sendWelcomeEmail({ email, first_name })
```
Comment (royanger) [inline docs/integrations/webhooks/inngest.mdx:132]: I believe that `emails` is just a placeholder?
```suggestion
   // `emails` is a placeholder for the function in your codebase to send email
   await emails.sendWelcomeEmail({ email, first_name })
```

File: docs/integrations/webhooks/inngest.mdx [hunk 17]
```diff
@@ -0,0 +1,215 @@
...
+ { event: 'clerk/user.created' },
+ async ({ event, step }) => { // ← step is available in the handler's arguments
+   const { user } = event.data
+   const { first_name } = user
+   const email = user.email_addresses.find(e =>
+     e.id === user.primary_email_address_id
+   ).email
+
+   await step.run('welcome-email', async () => {
+     await emails.sendWelcomeEmail({ email, first_name })
```
Comment (royanger) [inline docs/integrations/webhooks/inngest.mdx:159]:
```suggestion
     // `emails` is a placeholder for the function in your codebase to send email
     await emails.sendWelcomeEmail({ email, first_name })
```

File: docs/integrations/webhooks/inngest.mdx [hunk 18]
```diff
@@ -0,0 +1,215 @@
...
+
+During local development with Inngest, you can use the [Inngest Dev Server](https://www.inngest.com/docs/local-development) to run and test your functions on your own machine. To start the server, in your project directory run the following command:
+
+```sh filename="terminal"
+npx inngest-cli@latest dev
+```
+
+In your browser open [http://localhost:8288](http://localhost:8288) to see the development UI.
+
+To quickly get events to test within Dev Server, you can select any individual event from the **Events** tab then select the **Send to Dev Server**.
```
Comment (royanger) [inline docs/integrations/webhooks/inngest.mdx:186]: It wasn't super clear here that this step was back in the Inngest dashboard.

**Review by royanger — COMMENTED** [review]
This looks great! I dropped a few thoughts and suggestions below, but excited to see this live in the docs.

File: docs/integrations/webhooks/inngest.mdx [hunk 19]
```diff
@@ -0,0 +1,215 @@
...
+
+* [Limiting concurrency](https://www.inngest.com/docs/guides/concurrency) to handle spikes in events without overwhelming your API or database.
+* Triggering multiple functions from a single event ([fan-out jobs](https://www.inngest.com/docs/guides/fan-out-jobs)).
+* [Delaying code](https://www.inngest.com/docs/functions/multi-step) to run after a period of time.
+* [Debouncing events](https://www.inngest.com/docs/reference/functions/debounce) to reduce duplicate processing.
+
+In this guide, you'll learn how to set up Inngest to receive Clerk webhook events and how to define Inngest functions in your application using Clerk events including synchronizing data and sending welcome emails.
+
+To follow this guide, you need an Inngest account and have Inngest [set up](https://www.inngest.com/docs/quick-start) in your codebase.
```
Comment (sylwiavargas) [inline docs/integrations/webhooks/inngest.mdx:17]:
```suggestion
To follow this guide, you need an Inngest account (free tier is enough) and have Inngest [set up](https://www.inngest.com/docs/quick-start) in your codebase.
```

File: docs/integrations/webhooks/inngest.mdx (hunk 12 above)
Comment (sylwiavargas) [inline docs/integrations/webhooks/inngest.mdx:72]:
```suggestion
After setup, as webhook events are sent from Clerk to Inngest, new `clerk` events will appear in your [Inngest dashboard](https://app.inngest.com/env/production/events). Event names will be the prefixed `clerk/` followed by the event name. See [webhook events](/docs/integrations/webhooks#supported-webhook-events) for a full list.
```

File: docs/integrations/webhooks/inngest.mdx (hunk 18 above)
Comment (sylwiavargas) [reply docs/integrations/webhooks/inngest.mdx:187]: I've just pushed a commit to the previous paragraph. Do you think it clarifies that we are now operating on the Dev Server UI? (It's not the Inngest Dashboard, it's local development)

File: docs/integrations/webhooks/inngest.mdx (hunk 15 above)
Comment (sylwiavargas) [reply docs/integrations/webhooks/inngest.mdx:146]: That’s a good point. It makes sense to decouple either big tasks or unrelated tasks. Here, one task results from another and they constitute one whole.

I'll rework the previous section a little to make this clearer!

File: docs/integrations/webhooks/inngest.mdx [hunk 20]
```diff
@@ -0,0 +1,215 @@
...
+As you can see, you can choose which events you want to handle with each function. You might write a separate function for `clerk/user.updated` and `clerk/user.deleted` handling the entire lifecycle end to end.
+
+Note that multiple functions can also listen to the same event. This pattern is called “[fan-out](https://www.inngest.com/docs/guides/fan-out-jobs).”
+
+
+## Creating a function to send a welcome email
+
+Often, applications need to perform additional tasks when a new user is created, like send a welcome email with tips and useful information.
+
+While it is possible to add this logic at the end of your sync function as seen in the [previous section](/docs/integrations/webhooks/inngest#creating-a-function-to-sync-a-new-user-to-a-database), it's better to decouple logic into different functions so issues with one task do not affect the other ones. For example, if your email fails to send, it should not affect adding the user to an email list.
```
Comment (sylwiavargas) [inline docs/integrations/webhooks/inngest.mdx:115]:
```suggestion
While it is possible to add this logic at the end of your sync function as seen in the [previous section](/docs/integrations/webhooks/inngest#creating-a-function-to-sync-a-new-user-to-a-database), it’s better to decouple unrelated tasks into different functions so issues with one task do not affect the other ones. For example, if your email fails to send, it should not affect starting a trial for that user in Stripe.
```

Comment (sylwiavargas) [inline docs/integrations/webhooks/inngest.mdx]: @royanger let me know if this paragraph is now clearer 🙏

File: docs/integrations/webhooks/inngest.mdx (hunk 15 above)
Comment (sylwiavargas) [reply docs/integrations/webhooks/inngest.mdx:146]: See https://github.com/clerk/clerk-docs/pull/645/files/388815e0218521951e3e4dc48ebc67989798c6a9#r1459451471

File: docs/integrations/webhooks/inngest.mdx [hunk 21]
```diff
@@ -0,0 +1,215 @@
...
+  height={860}
+  src="/docs/images/integrations/inngest/inngest-clerk-events.webp"
+  alt="The Events page in the Inngest Dashboard showing a list of Clerk events."
+/>
+
+## Creating a function to sync a new user to a database
+
+With Inngest already set up in your codebase, you can now use these events as triggers for your functions.
+
+The code below features the `inngest.createFunction` method, which inserts a new user into the database. It will be triggered whenever `clerk/user.created` event occurs.
```
Comment (sylwiavargas) [inline docs/integrations/webhooks/inngest.mdx:85]:
```suggestion
Suppose you need to write a function which will insert a new user into the database which will be triggered whenever `clerk/user.created` event occurs. You would use the `inngest.createFunction` method, like in the example below:
```

File: docs/integrations/webhooks/inngest.mdx [hunk 22]
```diff
@@ -0,0 +1,215 @@
...
+
+
+```typescript
```
Comment (sylwiavargas) [inline docs/integrations/webhooks/inngest.mdx:88]:
```suggestion
```typescript filename="src/inngest/sync-user.ts"
```

File: docs/integrations/webhooks/inngest.mdx (hunk 13 above)
Comment (sylwiavargas) [reply docs/integrations/webhooks/inngest.mdx]: Good idea and feedback. I've changed the description paragraph, added the file name, and a comment about the `id` ✅ Thank you!

**Comment by royanger** [comment]
> _Additional context_: This guide may be more appropriate along side the "Sync Data" guide in the users section as it shows how a developer can use Clerk webhooks for syncing, but a lot more. I decided to submit in the integrations section as that's also where the info for Webhooks is. Very happy to make adjustments or link to this guide from the "Sync Data" guide as well.

My initial thought was that this fits in the general Guides section, but I saw the Inngest template in the Webhooks section and I think that makes it an Integration. But it could go either way, in my opinion.

File: docs/integrations/webhooks/inngest.mdx (hunk 20 above)
Comment (royanger) [reply docs/integrations/webhooks/inngest.mdx]: Looks great! Feel free to resolve this once you see!

File: docs/integrations/webhooks/inngest.mdx (hunk 18 above)
Comment (royanger) [reply docs/integrations/webhooks/inngest.mdx:188]: This definitely better. I was confused -- I ran the dev server but didn't really do anything with it. It looked different on its starting screen so I thought the following step was back in the Dashboard. Great clarification for someone just reading!

Please resolve once you read this.

File: docs/integrations/webhooks/inngest.mdx [hunk 23]
```diff
@@ -0,0 +1,217 @@
...
+Often, applications need to perform additional tasks when a new user is created, like send a welcome email with tips and useful information.
+
+While it is possible to add this logic at the end of your sync function as seen in the [previous section](/docs/integrations/webhooks/inngest#creating-a-function-to-sync-a-new-user-to-a-database), it’s better to decouple unrelated tasks into different functions so issues with one task do not affect the other ones. For example, if your email fails to send, it should not affect starting a trial for that user in Stripe.
+
+With Inngest, each function has automatic retries, so only the code that has issues is re-run.
+
+The code below creates another function using the same `clerk/user.created` event and adds the logic to send the welcome email:
+
+
+```typescript
```
Comment (royanger) [inline docs/integrations/webhooks/inngest.mdx:122]:
```suggestion
```typescript  filename="src/inngest/welcome-emails.ts"
```

File: docs/integrations/webhooks/inngest.mdx [hunk 24]
```diff
@@ -0,0 +1,217 @@
...
+Now, you have a function that utilizes the same Clerk webhook event for another purpose. Clerk webhook events can be used for all sorts of application lifecycle use cases. For example, adding users to a marketing email list, starting a Stripe trial, or provisioning new account resources.
+
+
+### Sending a delayed follow-up email
+
+Every Inngest function handler has an additional `step` object which provides tools to create more complex functions. Using `step.run` allows you to encapsulate specific code that will be automatically retried ensuring that issues with one part of your function don't force the entire function to re-run. Additionally, [other tools like `step.sleep`](https://www.inngest.com/docs/reference/functions/step-sleep), are available to extend functionality.
+
+The code below sends a welcome email, then uses `step.sleep` to wait for three days before sending another email offering a free trial:
+
+```typescript
```
Comment (royanger) [inline docs/integrations/webhooks/inngest.mdx:148]:
```suggestion
```typescript filename="src/inngest/welcome-emails.ts"
```

**Review by royanger — CHANGES_REQUESTED** [review]
Thanks for the changes!

Just two minor ones to add more filenames, and then info and an image for the Overview page.

Please see https://clerk.com/docs/integrations/overview for reference and then shoot me a message with the svg and the one sentence description and I will add them. Its in a private repo.

**Comment by djfarrelly** [comment]
Hey @royanger, thanks - here's the copy and logos (color and white) for the top level page:

> Use events from Clerk to trigger functions defined in your codebase with Inngest. Easily handle data synchronization, onboarding campaigns, or billing workflows in code.

inngest-logo-black
inngest-logo-white

Thanks and let us know if you need anything else!

**Review by domitriusclark — APPROVED** [review]

**Review by domitriusclark — APPROVED** [review]

**Comment by royanger** [comment]
Hey @djfarrelly. The guide itself is merged. I have a preview of the Integrations Overview. We need a new logo for two reasons. First, the way this is setup we can't serve light or new -- we just need one for both. Two, I think it needs to be cropped. See the preview -- its very small. https://clerk-marketing-git-roy-add-inngest-to-integrations-over-ee195b.clerkstage.dev/docs/integrations/overview cc @sylwiavargas cc @domitriusclark

**Pull request merged**